src/integrators/connector.jl

```diff
@@ -1,23 +1,29 @@
 export Connector,getConnector
 
-mutable struct Connector{T,TQ <: Union{Array{T,2},UniformScaling{Int}}, TK <: Union{Array{T,2},SparseMatrixCSC{T,Int}}} <: AbstractMeganetElement{T}
+mutable struct Connector{T,TQ <: Union{Array{T,2},UniformScaling{Int}}, TK <: Union{Function, Array{T,2},SparseMatrixCSC{T,Int}}} <: AbstractMeganetElement{T}
     K::TK
+    KT::Nullable{Function}
     b::T
     outTimes::Int
     Q::TQ # ???
+    nFeatIn::Int
+    nFeatOut::Int
 end
 
 nTheta(this::Connector) = 0
-nFeatIn(this::Connector) = size(this.K,2)
-nFeatOut(this::Connector) = size(this.K,1)
+nFeatIn(this::Connector) = this.nFeatIn
+nFeatOut(this::Connector) = this.nFeatOut
 nDataOut(this::Connector) = ((this.Q==I) ? nFeatOut(this) : size(this.Q,1))
 initTheta(this::Connector{T}) where {T <: Number} = zeros(T,0)
 
-function getConnector(TYPE::Type, K; b = zero(TYPE),outTimes=0,Q=I)
-	return Connector(K,b,outTimes,Q);
+function getConnector(TYPE::Type, K::AbstractArray; b = zero(TYPE), outTimes=0, Q=I)
+	return Connector(K, Nullable{Function}(), b, outTimes, Q, size(K,2), size(K,1))
+end
+function getConnector(TYPE::Type, K::Function, KT::Function, nFeatIn::Int, nFeatOut::Int; b = zero(TYPE), outTimes=0, Q=I)
+	return Connector(K, Nullable{Function}(KT), b, outTimes, Q, nFeatIn, nFeatOut)
 end
 
-function apply(this::Connector{T},theta::Array{T},Y0::Array{T},tmp,doDerivative=true) where {T <: Number}
+function apply(this::Connector{T, <:Any, <:Any},theta::Array{T},Y0::Array{T},tmp,doDerivative=true) where {T <: Number}
```

Comment:
pretty sure you can leave it as this::Connector{T} since Connector{T, blah,blah} <: Connector{T}

Reply:
I thought so too, but it was sometimes dispatching to the other one